scripts/rollup.config.js

```diff
@@ -1,49 +1,140 @@
+  This configuration is not minimized and will import all dependencies.
+*/
+const libConfig = baseConfig();
+// Do not include any of the dependencies
+libConfig.external = peerDependencies.concat(dependencies);
+libConfig.output = [
+    { sourcemap: true, name: 'DesignReactKit', file: packageJson.module + '.cjs.js', format: 'cjs' },
+    { sourcemap: true, name: 'DesignReactKit', file: packageJson.module + '.es.js', format: 'es' },
+];
+
+/*
+  UMD CONFIG
+  ----------
+  Goal of this configuration is to be directly included on web pages.
+  This configuration is minimized and will include dependencies that are not
+  marked as peer dependencies. ** See below
+  Defining this config will also check that all peer dependencies are set up
+  correctly in the globals input.
+  Reactstrap has two versions:
+  1) `reactstrap.min.js`
+      This file excludes `react-popper` and `react-transition-group` from
+      the dist build where they need to be manually required if any
+      application uses components that require these features.
+  2) `reactstrap.full.min.js`
+      This file includes all dependencies.
+  For both versions the peer dependencies are always excluded and must be manually
+  included - `react` and `react-dom`.
+*/
+const umdFullConfig = baseUmdConfig(false);
+umdFullConfig.output = [
+    { globals: globals(), sourcemap: true, name: 'DesignReactKit', file: packageJson.module + '.full.js', format: 'umd' },
+];
+
+// Validate globals in main UMD config
+const missingGlobals = peerDependencies.filter(dep => !(dep in globals()));
+if (missingGlobals.length) {
+    console.error('All peer dependencies need to be mentioned in globals, please update rollup.config.js.');
+    console.error('Missing: ' + missingGlobals.join(', '));
+    console.error('Aborting build.');
+    process.exit(1);
+}
+
+const umdFullConfigMin = baseUmdConfig(true);
+umdFullConfigMin.output = [
+    { globals: globals(), sourcemap: true, name: 'DesignReactKit', file: packageJson.module + '.full.min.js', format: 'umd' },
+];
+
+const external = umdFullConfig.external.slice();
+external.push('react-transition-group/Transition');
+external.push('react-popper');
+
+const allGlobals = Object.assign({}, globals(), {
+    'react-popper': 'ReactPopper',
+    'react-transition-group/Transition': 'ReactTransitionGroup.Transition',
+});
+
+const umdConfig = baseUmdConfig(false);
+umdConfig.external = external;
+umdConfig.output = [
+    { globals: allGlobals, sourcemap: true, name: 'DesignReactKit', file: packageJson.module + '.js', format: 'umd' },
+];
+
+const umdConfigMin = baseUmdConfig(true);
+umdConfigMin.external = external;
+umdConfigMin.output = [
+    { globals: allGlobals, sourcemap: true, name: 'DesignReactKit', file: packageJson.module + '.min.js', format: 'umd' },
+];
+
+
+export default [
+    libConfig,
+    umdFullConfig,
+    umdFullConfigMin,
+    umdConfig,
+    umdConfigMin,
```

Comment:
This is just a rewrite inspired by `reactstrap`... still not sure everything works perfectly in any scenario.

Reply:
No worries, we can check that later!